### PR TITLE
feat: support for mermaid

### PR DESCRIPTION
**Pull request by jdevries3133**

### CODE REVIEW

File: src/mermaidRemarkPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    // Of course, taking a step back, we'd ideally want to compile our
+    // mermaid source into static content at build time, but the mermaid
+    // library is so heavily dependent on the DOM I don't see anything out
+    // there to accomplish this that doesn't user puppeteer / browser
+    // automation, which is just deeply cursed (hello mermaid-cli...).
+    //
+    // Only note that if we are ever in a position where we need/want to remove
+    // this, we should probably register an error-handler with Mermaid
+    // client-side, because the default mermaid, "syntax error," UI is very
+    // ugly, and is currently being effectively hidden!!
```
Comment (jdevries3133) [inline src/mermaidRemarkPlugin.ts:110]: Context on the patch-packages situation is here.

File: src/pages/designSystem.astro [hunk 2]
```diff
@@ -78,4 +79,7 @@ import Html from "../layouts/Html.astro";
       </div>
     </div>
   </div>
+  <div class="prose">
+    <MermaidDemo />
+  </div>
```
Comment (jdevries3133) [inline src/pages/designSystem.astro:84]: Go to `/designSystem` to see mermaid in action.

File: patches/mermaid+10.2.2.patch
Comment (jdevries3133) [inline patches/mermaid+10.2.2.patch:1]: see `mermaidRemarkPlugin.ts` for context here

**Comment by jdevries3133** [comment]
I did notice that mermaid charts can get weird on small screen sizes. I think I might like to merge that and leave this as an area for future improvement, though, unless you happen to know of a quick fix off-hand.

**Comment by royemosby** [comment]
With all these compounded challenges and the fact that you have to patch the package to get it operable, we should probably demote it to "nice to have". I'm invested in the notion of eventually using Mermaid so we can keep this around so we can revisit later.

**Comment by royemosby** [comment]
https://github.com/Code-the-Dream-School/school-pages/pull/65#issuecomment-1575197125

I see that and I think the mobile rendering really messes any chance of using it for now :-(

**Pull request merged**